### PR TITLE
Cascade delete samples when subject is deleted or unpublished

### PR DESCRIPTION
**Pull request by iamigo**

### CODE REVIEW

**Comment by shriramshankar** [comment]
Are we doing migrations and deleting the samples in cache as a part of a different story?

**Comment by iamigo** [comment]
> migrations

Will be a separate PR.

> deleting the samples in cache

Will add that to this PR today.

File: db/model/subject.js [hunk 1]
```diff
@@ -431,6 +425,11 @@ module.exports = function subject(seq, dataTypes) {
         } else if (inst.previous('isPublished')) {
           // Treat unpublishing a subject as a "delete" event.
           common.publishChange(inst, eventName.del);
+
+          if (featureToggles.isFeatureEnabled(sampleStoreFeature)) {
+            subjectUtils.removeFromRedis(inst.absolutePath);
+          }
```
Comment (shriramshankar) [inline db/model/subject.js]: Do we need this here too? I think the call to removeFromRedis at 354 will take care of this case.

Comment (iamigo) [inline db/model/subject.js]: That is true--just running tests now and will update in a moment

**Pull request merged**